profile/src/v100/entity/persona/persona_data/entry_kinds/name.rs

```diff
@@ -0,0 +1,171 @@
+use serde::{Deserialize, Serialize};
+use std::{cmp::Ordering, fmt::Display};
+
+use crate::prelude::*;
+
+#[derive(
+    Serialize,
+    Deserialize,
+    Clone,
+    Debug,
+    PartialEq,
+    Hash,
+    Eq,
+    uniffi::Record,
+    Default,
+)]
+pub struct Name {
+    variant: Variant,
+    family_name: String,
+    given_name: String,
+    nickname: String,
+}
+
+impl Name {
+    pub fn new(
+        variant: Variant,
+        family_name: &str,
```

Comment:
Not sure I fully understood how to implement exactly what you're after here. This feels somewhat off since I'm taking a parameter that `impl AsRef<str>`. Then I convert that value to a ref before finally converting it back to a String.

```
pub fn new(
        variant: Variant,
        family_name: impl AsRef<str>,
        given_name: impl AsRef<str>,
        nickname: impl AsRef<str>,
    ) -> Self {
        Self {
            variant,
            family_name: family_name.as_ref().to_string(),
            given_name: given_name.as_ref().to_string(),
            nickname: nickname.as_ref().to_string(),
        }
    }
```

Reply:
This is correct:)